### PR TITLE
Enhancement: Comprehensive Permission Management(RBAC)

### PR DESCRIPTION
**Pull request by faid-terence**
#### What does this PR do?
This PR introduces comprehensive permission management functionality, encompassing CRUD (Create, Read, Update, Delete) operations for permissions, along with the capability to assign these permissions to specific roles
#### Description of Task to be completed?
The primary objective of this task is to implement a comprehensive permission management system that includes CRUD operations for individual permissions and the ability to assign these permissions to specific roles within the application.
#### How should this be manually tested?

```
git clone  https://github.com/techly-africa/mpa-cash-backend.git    
cd mpacash-backend
git checkout branch-feature/permission-management
npm i
npm run start:dev
```

#### What are the relevant pivotal tracker/Trello/Asana stories?

Trello [Link]() (#-?)

#### Screenshots (if appropriate)
Screenshot 2023-10-11 111459
Screenshot 2023-10-11 111651

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,79 @@
-<a href="https://codeclimate.com/github/techly-africa/mpa-cash-backend/maintainability"><img src="https://api.codeclimate.com/v1/badges/1f6c2024dcbec25991dc/maintainability" /></a>
-<a href='https://coveralls.io/github/techly-africa/mpa-cash-backend?branch=main'><img src='https://coveralls.io/repos/github/techly-africa/mpa-cash-backend/badge.svg?branch=main' alt='Coverage Status' /></a>
-<p align="center">
-  <a href="http://nestjs.com/" target="blank"><img src="https://nestjs.com/img/logo-small.svg" width="200" alt="Nest Logo" /></a>
-</p>
+# Role-Based Access Control Backend(Auth-X)
```
Comment (edwinnambaje) [inline README.md:1]: could you re add the coverage badge from coveralls?

Comment (faid-terence) [reply README.md:1]: Yes @edwinnambaje 
The Badge is there but i think we need to Update it since the repo's name has changed

Comment (faid-terence) [reply README.md:1]: If you have a new one please share it with me and i update it 
Thanks

Comment (edwinnambaje) [reply README.md:1]: i will add it there no problem

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Pull request merged**